Build failure: Openbabel

### Steps To Reproduce

Steps to reproduce the behavior:
1. build openbabel or any package that depends on it, i.e Kalzium

### Build log

```
┏━ 1 Errors: 
 ⋮ 
┃        > -- Set non-toolchain portion of runtime path of "/nix/store/h7kpjljvzr2zhy06jr1sz5p5wlslyl0d-openbabel-unstable-06-12-23/bin/obspectrophore" to "/nix/store
/h7kp…
┃        > -- Installing: /nix/store/h7kpjljvzr2zhy06jr1sz5p5wlslyl0d-openbabel-unstable-06-12-23/bin/obrms
┃        > -- Set non-toolchain portion of runtime path of "/nix/store/h7kpjljvzr2zhy06jr1sz5p5wlslyl0d-openbabel-unstable-06-12-23/bin/obrms" to "/nix/store/h7kpjljv
zr2zh…
┃        > -- Installing: /nix/store/h7kpjljvzr2zhy06jr1sz5p5wlslyl0d-openbabel-unstable-06-12-23/bin/roundtrip
┃        > -- Set non-toolchain portion of runtime path of "/nix/store/h7kpjljvzr2zhy06jr1sz5p5wlslyl0d-openbabel-unstable-06-12-23/bin/roundtrip" to "/nix/store/h7kp
jljvz…
┃        > CMake Error at scripts/cmake_install.cmake:60 (file):
┃        >   file cannot create directory: /openbabel.  Maybe need administrative
┃        >   privileges.
┃        > Call Stack (most recent call first):
┃        >   cmake_install.cmake:115 (include)
┃        >
┃        > 
┃        > make: *** [Makefile:110: install] Error 1
┃        For full logs, run 'nix log /nix/store/hqmxz2my2zb8bpfkap7hw9afx2xwfv52-openbabel-unstable-06-12-23.drv'.

```

### Additional context

The issue seems to have appeared in the most recent push to nixos-unstable, it seems to be an issue with the openbabel dependency

### Notify maintainers

@danielbarter 

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.8-cachyos, NixOS, 24.11 (Vicuna), 24.11.20240708.655a58a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.23.2`
 - nixpkgs: `/nix/store/dk2rpyb6ndvfbf19bkb2plcz5y3k8i5v-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/openbabel/default.nix
+++ b/pkgs/development/libraries/openbabel/default.nix
@@ -12,6 +12,9 @@
 , pkg-config
 , swig
 , rapidjson
+, boost
+, maeparser
+, coordgenlibs
 }:
 
 stdenv.mkDerivation rec {
@@ -29,7 +32,7 @@ stdenv.mkDerivation rec {
     sed '1i#include <ctime>' -i include/openbabel/obutil.h # gcc12
   '';
 
-  buildInputs = [ perl zlib libxml2 eigen python cairo pcre swig rapidjson ];
+  buildInputs = [ perl zlib libxml2 eigen python cairo pcre swig rapidjson boost maeparser coordgenlibs ];
 
   nativeBuildInputs = [ cmake pkg-config ];
 

--- a/pkgs/development/libraries/openbabel/default.nix
+++ b/pkgs/development/libraries/openbabel/default.nix
@@ -1,4 +1,18 @@
-{ stdenv, lib, fetchFromGitHub, cmake, perl, zlib, libxml2, eigen, python, cairo, pcre, pkg-config, swig, rapidjson }:
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, perl
+, zlib
+, libxml2
+, eigen
+, python
+, cairo
+, pcre
+, pkg-config
+, swig
+, rapidjson
+}:
 
 stdenv.mkDerivation rec {
   pname = "openbabel";
@@ -19,18 +33,19 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake pkg-config ];
 
-  pythonMajorMinor = "${python.sourceVersion.major}.${python.sourceVersion.minor}";
-
-  cmakeFlags = [
-    "-DRUN_SWIG=ON"
-    "-DPYTHON_BINDINGS=ON"
-  ];
+  preConfigure = ''
+    cmakeFlagsArray+=(
+      "-DRUN_SWIG=ON"
+      "-DPYTHON_BINDINGS=ON"
+      "-DPYTHON_INSTDIR=$out/${python.sitePackages}"
+    )
+  '';
 
   # Setuptools only accepts PEP 440 version strings. The "unstable" identifier
   # can not be used. Instead we pretend to be the 3.2 beta release.
   postFixup = ''
-    cat <<EOF > $out/lib/python$pythonMajorMinor/site-packages/setup.py
-    from distutils.core import setup
+    cat << EOF > $out/${python.sitePackages}/setup.py
+    from setuptools import setup
 
     setup(
         name = 'pyopenbabel',

--- a/pkgs/development/python-modules/openbabel-bindings/default.nix
+++ b/pkgs/development/python-modules/openbabel-bindings/default.nix
@@ -8,9 +8,9 @@
 buildPythonPackage rec {
   inherit (openbabel) pname version;
 
-  src = "${openbabel}/lib/python${python.sourceVersion.major}.${python.sourceVersion.minor}/site-packages";
+  src = "${openbabel}/${python.sitePackages}";
 
-  nativeBuildInputs = [ openbabel ];
+  buildInputs = [ openbabel ];
 
   # these env variables are used by the bindings to find libraries
   # they need to be included explicitly in your nix-shell for

